flink-core/src/main/java/org/apache/flink/api/common/serialization/SerializerConfigImpl.java

```diff
@@ -356,6 +356,9 @@ public void configure(ReadableConfig configuration, ClassLoader classLoader) {
         configuration
                 .getOptional(PipelineOptions.FORCE_KRYO_AVRO)
                 .ifPresent(this::setForceKryoAvro);
+        configuration
+                .getOptional(PipelineOptions.BUILT_IN_COLLECTION_TYPES)
+                .ifPresent(TypeExtractor::setBuiltInCollectionTypesEnabled);
```

Comment:
The configuring approach seems problematic. As far as I know, the TypeExtractor is a static util class shared in a TaskManager while the SerializerConfig is scoped to the job level. This may lead to issues when multiple jobs with different config values are submitted to the same cluster.